author	Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
commit	9d656c7a1d60b5d8f2ceeb9734f453677259fb9a
	Fri, 29 Jun 2012 11:07:31 +0000 (14:07 +0300)

AVDTP: Remove auto_dc flag

auto_dc flag is no longer useful as all the users of it just reset it
to FALSE, also this was first introduced to maintain the stream for
some time as the clients disconnected frequently, but this is not the
case anymore.

diff -cr bluez-4.99.pan.orig/audio/avdtp.c bluez-4.99.pan/audio/avdtp.c
*** bluez-4.99.pan.orig/audio/avdtp.c	Tue Oct 16 09:51:05 2012
--- bluez-4.99.pan/audio/avdtp.c	Tue Oct 16 15:42:32 2012
***************
*** 395,403 ****
  
  	avdtp_session_state_t state;
  
- 	/* True if the session should be automatically disconnected */
- 	gboolean auto_dc;
- 
  	/* True if the entire device is being disconnected */
  	gboolean device_disconnect;
  
--- 395,400 ----
***************
*** 805,823 ****
  	g_free(stream);
  }
  
- static gboolean stream_timeout(gpointer user_data)
- {
- 	struct avdtp_stream *stream = user_data;
- 	struct avdtp *session = stream->session;
- 
- 	if (avdtp_close(session, stream, FALSE) < 0)
- 		error("stream_timeout: closing AVDTP stream failed");
- 
- 	stream->idle_timer = 0;
- 
- 	return FALSE;
- }
- 
  static gboolean transport_cb(GIOChannel *chan, GIOCondition cond,
  				gpointer data)
  {
--- 802,807 ----
***************
*** 1068,1078 ****
  		break;
  	case AVDTP_STATE_OPEN:
  		stream->starting = FALSE;
- 		if ((old_state > AVDTP_STATE_OPEN && session->auto_dc) ||
- 							stream->open_acp)
- 			stream->idle_timer = g_timeout_add_seconds(STREAM_TIMEOUT,
- 								stream_timeout,
- 								stream);
  		break;
  	case AVDTP_STATE_STREAMING:
  		if (stream->idle_timer) {
--- 1052,1057 ----
***************
*** 1196,1203 ****
  	if (session->dc_timer)
  		remove_disconnect_timer(session);
  
- 	session->auto_dc = TRUE;
- 
  	if (session->ref != 1)
  		error("connection_lost: ref count not 1 after all callbacks");
  	else
--- 1175,1180 ----
***************
*** 2394,2400 ****
  	/* We don't use avdtp_set_state() here since this isn't a state change
  	 * but just setting of the initial state */
  	session->state = AVDTP_SESSION_STATE_DISCONNECTED;
- 	session->auto_dc = TRUE;
  
  	session->version = get_version(session);
  
--- 2371,2376 ----
***************
*** 2467,2476 ****
  						(GIOFunc) session_cb, session,
  						NULL);
  
! 		if (session->stream_setup) {
  			set_disconnect_timer(session);
- 			avdtp_set_auto_disconnect(session, FALSE);
- 		}
  	} else if (session->pending_open)
  		handle_transport_connect(session, chan, session->imtu,
  								session->omtu);
--- 2443,2450 ----
  						(GIOFunc) session_cb, session,
  						NULL);
  
! 		if (session->stream_setup)
  			set_disconnect_timer(session);
  	} else if (session->pending_open)
  		handle_transport_connect(session, chan, session->imtu,
  								session->omtu);
***************
*** 4008,4018 ****
  	return g_slist_find(session->streams, stream) ? TRUE : FALSE;
  }
  
- void avdtp_set_auto_disconnect(struct avdtp *session, gboolean auto_dc)
- {
- 	session->auto_dc = auto_dc;
- }
- 
  gboolean avdtp_stream_setup_active(struct avdtp *session)
  {
  	return session->stream_setup;
--- 3982,3987 ----
diff -cr bluez-4.99.pan.orig/audio/avdtp.h bluez-4.99.pan/audio/avdtp.h
*** bluez-4.99.pan.orig/audio/avdtp.h	Tue Oct 16 09:51:05 2012
--- bluez-4.99.pan/audio/avdtp.h	Tue Oct 16 15:43:06 2012
***************
*** 308,314 ****
  
  void avdtp_get_peers(struct avdtp *session, bdaddr_t *src, bdaddr_t *dst);
  
- void avdtp_set_auto_disconnect(struct avdtp *session, gboolean auto_dc);
  gboolean avdtp_stream_setup_active(struct avdtp *session);
  void avdtp_set_device_disconnect(struct avdtp *session, gboolean dev_dc);
  
--- 308,313 ----
diff -cr bluez-4.99.pan.orig/audio/sink.c bluez-4.99.pan/audio/sink.c
*** bluez-4.99.pan.orig/audio/sink.c	Mon Mar  5 04:03:57 2012
--- bluez-4.99.pan/audio/sink.c	Tue Oct 16 15:43:29 2012
***************
*** 415,422 ****
  	if (!sink->session)
  		return FALSE;
  
- 	avdtp_set_auto_disconnect(sink->session, FALSE);
- 
  	if (avdtp_discover(sink->session, discovery_complete, sink) < 0)
  		return FALSE;
  
--- 415,420 ----
diff -cr bluez-4.99.pan.orig/audio/source.c bluez-4.99.pan/audio/source.c
*** bluez-4.99.pan.orig/audio/source.c	Mon Mar  5 04:03:57 2012
--- bluez-4.99.pan/audio/source.c	Tue Oct 16 15:43:54 2012
***************
*** 366,373 ****
  	if (!source->session)
  		return FALSE;
  
- 	avdtp_set_auto_disconnect(source->session, FALSE);
- 
  	if (avdtp_discover(source->session, discovery_complete, source) < 0)
  		return FALSE;
  
--- 366,371 ----
